docs/src/components/style-spec/specs/item.jsx

```diff
@@ -0,0 +1,370 @@
+                                    this.props.expression.parameters.includes(
+                                        'feature-state'
+                                    ) &&
+                                    ' and '}
+                                {this.props.expression.interpolated && (
+                                    <a href="/maplibre-gl-style-spec/style-spec/expressions/#interpolate">
+                                        {/* <Icon
+                                            name="smooth-ramp"
+                                            inline={true}
+                                        /> */}
+                                        <code>interpolate</code>
+                                    </a>
+                                )}
+                                expressions.{' '}
+                            </React.Fragment>
+                        )}
+
+                    {this.props.transition && (
+                        <React.Fragment>
+                            {/* <Icon name="opacity" inline={true} /> */}
+                            Transitionable.{' '}
+                        </React.Fragment>
+                    )}
+                </Subtitle>
+
+                {this.props.doc && (
+                    <div className="mb12 style-spec-item-doc">
+                        <ReactMarkdown>{this.props.doc}</ReactMarkdown>
+                    </div>
+                )}
+
+                {this.props.values &&
+                    !Array.isArray(this.props.values) && ( // skips $root.version
+                        <div className="my12 style-spec-item-dl">
+                            <dl>
+                                {entries(this.props.values).map(
+                                    ([v, { doc }], i) => [
+                                        <dt key={`${i}-dt`}>
+                                            <code>{JSON.stringify(v)}</code>:
+                                        </dt>,
+                                        <dd key={`${i}-dd`} className="mb12">
+                                                <ReactMarkdown>{doc}</ReactMarkdown>
+                                        </dd>
+                                    ]
+                                )}
+                            </dl>
+                        </div>
+                    )}
+
+                {this.props.example &&
+                    highlightJSON(
+                        `"${this.props.name}": ${JSON.stringify(
+                            this.props.example,
+                            null,
+                            2
+                        )}`
+                    )}
+
+                {this.props['sdk-support'] && (
+                    <div className="mt12">
+                        <SDKSupportTable {...this.props['sdk-support']} />
+                    </div>
+                )}
+            </React.Fragment>
+        );
+    }
+}
+
+Item.propTypes = {
```

Comment:
Can we use typescript instead of PropTypes?

Reply:
Runtime type checking is not so meaningful for SSG sites where everything happens at build-time, whereas improved static-analysis can be very helpful.

Reply:
yes

Reply:
ohh okay, I'll have to make all files tsx then

Reply:
actually I don;t have much experience with typescript tbh

Reply:
It's a superset of js so changing the files tsx should be correct syntax still

Reply:
Hi yeah I tried it, but it is giving me following type of errors
<img width="1174" alt="Screenshot 2023-03-07 at 11 33 29 AM" src="https://user-images.githubusercontent.com/47075664/223334335-a2e58987-a55f-47aa-81ca-e88f65ebd91f.png">

I'm guessing these are the errors related to typescript data type definitions right?

Reply:
Yes props need an interface like this

https://www.benmvp.com/blog/react-prop-types-with-typescript/